gazebo_ros/src/gazebo_ros_api_plugin.cpp

```diff
@@ -1655,9 +1649,7 @@ bool GazeboRosApiPlugin::setLinkState(gazebo_msgs::SetLinkState::Request &req,
     gazebo::math::Quaternion frame_rot = frame_pose.rot;
 
     //std::cout << " debug : " << frame->GetName() << " : " << frame_pose << " : " << target_pose << std::endl;
-    //target_pose = frame_pose + target_pose; // seems buggy, use my own
-    target_pose.pos = frame_pos + frame_rot.RotateVector(target_pos);
-    target_pose.rot = frame_rot * target_pose.rot;
+    target_pose = target_pose + frame_pose;
```

Comment:
Same as above. Shouldn't this be the other way around. I haven't tested this myself.